src/services/itemMembership/repository.ts

```diff
@@ -97,26 +97,22 @@ export const ItemMembershipRepository = AppDataSource.getRepository(ItemMembersh
   },
 
   async getForManyItems(items: Item[], memberId?: UUID): Promise<ResultOf<ItemMembership[]>> {
-    const query = this.createQueryBuilder('item_membership')
-      .leftJoinAndSelect('item_membership.item', 'item')
-      .leftJoinAndSelect('item_membership.member', 'member');
-
-    query.where(
-      new Brackets((qb) => {
-        items.forEach(({ path }, idx) => {
-          // if (idx === 0) {
-          //   qb.where(`item.path @> :path_${path}`, { [`path_${path}`]: path });
-          // } else {
-          qb.orWhere(`item.path @> :path_${path}`, { [`path_${path}`]: path });
-          // }
-        });
-      }),
-    );
+    if (items.length === 0) {
+      return { data: {}, errors: [] };
+    }
 
+    const ids = items.map((i) => i.id);
+    const query = this.createQueryBuilder('item_membership')
+      .innerJoin('item', 'descendant', 'item_membership.item_path @> descendant.path')
+      .where('descendant.id in (:...ids)', { ids: ids });
     if (memberId) {
       query.andWhere('member.id = :memberId', { memberId });
     }
 
+    query
+      .leftJoinAndSelect('item_membership.item', 'item')
```

Comment:
Could you exchange `innerJoin` to `innerJoinAndSelect` and remove this line?

Reply:
No this is not the same join, the first one is just used as a filter to retrieve memberships that are parents, but this is not a join where `item_membership.item_path = item.path`. The second one is for filling the entity that will be returned with the real linked item
I.E: I need to do the two join, at least from what I was trying to do